lib/ts/BaseLogFile.cc

```diff
@@ -0,0 +1,578 @@
+ * Similar to above constructor, but is overloaded with the object signature
+ */
+BaseLogFile::BaseLogFile(const char *name, uint64_t sig) : m_signature(sig), m_has_signature(true)
+{
+  init(name);
+  log_log_trace("exiting BaseLogFile signature constructor, m_name=%s, m_signature=%ld, this=%p\n", (char *)m_name, m_signature,
+                this);
+}
+
+/*
+ * This copy constructor creates a BaseLogFile based on a given copy.
+ */
+BaseLogFile::BaseLogFile(const BaseLogFile &copy)
+  : m_fp(NULL), m_start_time(copy.m_start_time), m_end_time(0L), m_bytes_written(0), m_signature(copy.m_signature),
+    m_has_signature(copy.m_has_signature), m_name(ats_strdup(copy.m_name)), m_hostname(ats_strdup(copy.m_hostname)),
+    m_is_regfile(false), m_is_init(copy.m_is_init), m_meta_info(NULL)
+{
+  log_log_trace("exiting BaseLogFile copy constructor, m_name=%s, this=%p\n", (char *)m_name, this);
+}
+
+/*
+ * Destructor.
+ */
+BaseLogFile::~BaseLogFile()
+{
+  log_log_trace("entering BaseLogFile destructor, m_name=%s, this=%p\n", (char *)m_name, this);
+
+  if (m_is_regfile)
+    close_file();
+  else
+    log_log_trace("not a regular file, not closing, m_name=%s, this=%p\n", (char *)m_name, this);
+
+  log_log_trace("exiting BaseLogFile destructor, this=%p\n", this);
+}
+
+/*
+ * Initializes the defaults of some of the common member values of this class
+ */
+void
+BaseLogFile::init(const char *name)
+{
+  m_fp = NULL;
+  m_start_time = time(0);
+  m_end_time = 0L;
+  m_bytes_written = 0;
+  m_name = ats_strdup(name);
+  m_hostname = NULL;
+  m_is_regfile = false;
+  m_is_init = false;
+  m_meta_info = NULL;
+}
+
+/*
+ * This function is called by a client of BaseLogFile to roll the underlying
+ * file  The tricky part to this routine is in coming up with the new file name,
+ * which contains the bounding timestamp interval for the entries
+ * within the file.
+
+ * Under normal operating conditions, this BaseLogFile object was in existence
+ * for all writes to the file.  In this case, the LogFile members m_start_time
+ * and m_end_time will have the starting and ending times for the actual
+ * entries written to the file.
+
+ * On restart situations, it is possible to re-open an existing BaseLogFile,
+ * which means that the m_start_time variable will be later than the actual
+ * entries recorded in the file.  In this case, we'll use the creation time
+ * of the file, which should be recorded in the meta-information located on
+ * disk.
+
+ * If we can't use the meta-file, either because it's not there or because
+ * it's not valid, then we'll use timestamp 0 (Jan 1, 1970) as the starting
+ * bound.
+
+ * Return 1 if file rolled, 0 otherwise
+ */
+int
+BaseLogFile::roll(long interval_start, long interval_end)
+{
+  // First, let's see if a roll is even needed.
+  if (m_name == NULL || !BaseLogFile::exists((char *)m_name)) {
+    log_log_trace("Roll not needed for %s; file doesn't exist\n", ((char *)m_name) ? (char *)m_name : "no_name\n");
```

Comment:
For scoped strings, you only need to be explicit for macros, not for functions. Most of the time just "m_name" should work. In cases like this, you should use the `get` method instead of casting. E.g.
`log_log_trace("Roll not needed for %s; file doesn't exist\n", m_name ? m_name,get() : "no_name\n")